test/common/upstream/logical_dns_cluster_test.cc

```diff
@@ -88,13 +88,67 @@ TEST_F(LogicalDnsClusterTest, ImmediateResolve) {
   )EOF";
 
   EXPECT_CALL(initialized_, ready());
-  EXPECT_CALL(dns_resolver_, resolve("foo.bar.com", _))
-      .WillOnce(Invoke([&](const std::string&, Network::DnsResolver::ResolveCb cb)
-                           -> Network::ActiveDnsQuery* {
-                             EXPECT_CALL(*resolve_timer_, enableTimer(_));
-                             cb(TestUtility::makeDnsResponse({"127.0.0.1", "127.0.0.2"}));
-                             return nullptr;
-                           }));
+  EXPECT_CALL(dns_resolver_, resolve("foo.bar.com", Network::DnsLookupFamily::V4_ONLY, _))
+      .WillOnce(Invoke([&](const std::string&, const Network::DnsLookupFamily&,
+                           Network::DnsResolver::ResolveCb cb) -> Network::ActiveDnsQuery* {
+        EXPECT_CALL(*resolve_timer_, enableTimer(_));
+        cb(TestUtility::makeDnsResponse({"127.0.0.1", "127.0.0.2"}));
+        return nullptr;
+      }));
+  setup(json);
+  EXPECT_EQ(1UL, cluster_->hosts().size());
+  EXPECT_EQ(1UL, cluster_->healthyHosts().size());
+  EXPECT_EQ("foo.bar.com", cluster_->hosts()[0]->hostname());
+  tls_.shutdownThread();
+}
+
+TEST_F(LogicalDnsClusterTest, ImmediateResolveV6Only) {
+  std::string json = R"EOF(
+  {
+    "name": "name",
+    "connect_timeout_ms": 250,
+    "type": "logical_dns",
+    "lb_type": "round_robin",
+    "dns_lookup_family": "v6_only",
+    "hosts": [{"url": "tcp://foo.bar.com:443"}]
+  }
+  )EOF";
+
+  EXPECT_CALL(initialized_, ready());
+  EXPECT_CALL(dns_resolver_, resolve("foo.bar.com", Network::DnsLookupFamily::V6_ONLY, _))
+      .WillOnce(Invoke([&](const std::string&, const Network::DnsLookupFamily&,
+                           Network::DnsResolver::ResolveCb cb) -> Network::ActiveDnsQuery* {
+        EXPECT_CALL(*resolve_timer_, enableTimer(_));
+        cb(TestUtility::makeDnsResponse({"::1"}));
+        return nullptr;
+      }));
+  setup(json);
+  EXPECT_EQ(1UL, cluster_->hosts().size());
+  EXPECT_EQ(1UL, cluster_->healthyHosts().size());
+  EXPECT_EQ("foo.bar.com", cluster_->hosts()[0]->hostname());
+  tls_.shutdownThread();
+}
+
+TEST_F(LogicalDnsClusterTest, ImmediateResolveAuto) {
+  std::string json = R"EOF(
+  {
+    "name": "name",
+    "connect_timeout_ms": 250,
+    "type": "logical_dns",
+    "lb_type": "round_robin",
+    "dns_lookup_family": "auto",
+    "hosts": [{"url": "tcp://foo.bar.com:443"}]
+  }
+  )EOF";
+
+  EXPECT_CALL(initialized_, ready());
+  EXPECT_CALL(dns_resolver_, resolve("foo.bar.com", Network::DnsLookupFamily::AUTO, _))
+      .WillOnce(Invoke([&](const std::string&, const Network::DnsLookupFamily&,
+                           Network::DnsResolver::ResolveCb cb) -> Network::ActiveDnsQuery* {
+        EXPECT_CALL(*resolve_timer_, enableTimer(_));
+        cb(TestUtility::makeDnsResponse({"::1"}));
+        return nullptr;
+      }));
```

Comment:
There might be an argument for using `TEST_P` here, but I'm not sure it's worth the effort and it's probably more readable the way you've done it.